### ISSUE TITLE
LLVM 15 regression in C ABI

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#12683]
This issue regards the `llvm15` branch, with LLVM 15 rc3.

```
[nix-shell:~/Downloads/zig/test/c_abi]$ ../../build-llvm15-release/stage3/bin/zig build test
Test [11/16] test.C ABI big packed struct... thread 189888 panic: zig_panic called from C
/home/andy/Downloads/zig/test/c_abi/main.zig:8:5: 0x212e28 in zig_panic (test)
    @panic("zig_panic called from C");
    ^
/home/andy/Downloads/zig/test/c_abi/cfuncs.c:10:9: 0x2119b4 in assert_or_panic (/home/andy/Downloads/zig/test/c_abi/cfuncs.c)
        zig_panic();
        ^
/home/andy/Downloads/zig/test/c_abi/cfuncs.c:406:5: 0x212761 in c_big_packed_struct (/home/andy/Downloads/zig/test/c_abi/cfuncs.c)
    assert_or_panic(x.a == 1);
    ^
/home/andy/Downloads/zig/test/c_abi/main.zig:300:24: 0x2148ef in test.C ABI big packed struct (test)
    c_big_packed_struct(s);
                       ^
/home/andy/Downloads/zig/lib/test_runner.zig:79:28: 0x215d3a in main (test)
        } else test_fn.func();
                           ^
/home/andy/Downloads/zig/lib/std/start.zig:566:22: 0x2177ea in main (test)
            root.main();
                     ^
error: the following test command crashed:
/home/andy/Downloads/zig/test/c_abi/zig-cache/o/171c83f4affaff5df8cd360ed6498b4a/test /home/andy/Downloads/zig/build-llvm15-release/stage3/bin/zig
error: test...
error: The following command exited with error code 1:
/home/andy/Downloads/zig/build-llvm15-release/stage3/bin/zig test /home/andy/Downloads/zig/test/c_abi/main.zig /home/andy/Downloads/zig/test/c_abi/zig-cache/o/3e66e2cc9d26ac606f015847ed17fea8/cfuncs.o -lc --cache-dir /home/andy/Downloads/zig/test/c_abi/zig-cache --global-cache-dir /home/andy/.cache/zig --name test --enable-cache 
```

Likely related to bf28765a975355c27558eaa86cf00ccb29b663a7. I suspect that moving the parameter logic made it stop emitting the attribute for extern functions, causing the regression.

**Comment by Vexu** [comment]
The issue is that you moved adding the byval  attribute to `updateFunc` which isn't called for extern functions.

**Comment by andrewrk** [comment]
Thanks for the clue! Fixed in fb81ba87620467de11db6b904741b4aafc7cb756 - `byval` was not the only missing attribute.
